### ISSUE TITLE
SE Upgrade Database Issue

### ISSUE THREAD

**Issue by poeticjustice** [SocialEngine/phpv4-issues#724]
I just instaled the SE 4.9.1 upgrade, it confirmed ovewrite of files 
On the next page it displayed this message

No entry is registered for key 'Zend_Db' social engine

There was no other buttons to press, so pressed home, and it went to the initial install of social engine

asking for database credentials, and to name the network, and to choose defaul admin email

EVERYTHING IS GONE

All users deleted from MYsql, just deleted over a thousand users

Its wiped all profile questions, all networks, all profile types, all profile levels

This is my cloned staging server , with a cloned db btw, but if it was my live site...... id be more angry....... but serious what the hell

It identified 3 plugins were installed, but wiped all the others from the db

seriously w t f

@DonnaScriptTechs @Khushboo016

**Comment by poeticjustice** [comment]
All users deleted from MYsql

**Comment by poeticjustice** [comment]
Appears SE is deleting 3 files upon updating 

100% deleting /application/settings/database.php

On upgrade, tried it 7 times now, and that's definitely a file it deletes, this is why it goes thru the process of installing again, and wipes all the database of users

**Comment by poeticjustice** [comment]
I have videos, Its removing 3 files on upgrading
/application/settings/database.php                                                          (different_removed)
/application/settings/enabled_module_directories.php                           (different_removed)
/application/settings/general.php                                                             (different_removed)

**Comment by Elshara** [comment]
When did this started happening? I can't confirm this for upgrading. I
can however confirm some upgrading aspects aren't totally working as
they should be.
For instance, activity feed attachments.
This message seems to appear for the activity feed after upgrading.
en4.core.runonce.add(function () { en4.activity.bindEditFeed(16243, {
lang: { 'Post Something...': 'Post Something...' },
allowEmptyWithoutAttachment: 0 }); });
But nothing was done to suggest a complete upgrade override.
Scary.

**Comment by DonnaScriptTechs** [comment]
@poeticjustice please send a ticket to our support. We need to have a look at this. Please put the subject Github 724. Thank you.

**Comment by DonnaScriptTechs** [comment]
What version did you upgrade from?

**Comment by Elshara** [comment]
I upgraded fine from v4.8.13 and made every version change possibly
from v4.8.7 since then in order as they were released. So that's
probably why it went smoothly for me.

**Comment by DonnaScriptTechs** [comment]
I mean @poeticjustice , what version did you upgrade from?

**Comment by poeticjustice** [comment]
4.8.11

It keeps deleting them 3 files on upgrade

When I copy the deleted file back into the directory then change to sync up with cloned database, it's just full of errors, then in package manager it has loads of install options

It's weird

But if I didn't clone this site and did it to my live server , I'd be crying as it wipes the whole database during install period

I have videos

**Comment by DonnaScriptTechs** [comment]
Ray is looking into this issue. He says it appears a permission issue with some server configuration as we only have this one report. However, he is making a check so that file never gets deleted. 

A temp solution is to skip that file when given the option of files to skip.

**Comment by poeticjustice** [comment]
skipping will skip overwriting of all files tho? and there for update will not occur? @DonnaScriptTechs
Pretty sure you cant skip them 3 files.
As it shows all file being overwritten 

Do you want the video? so you can see exactly

all file permissions are set accordingly tho

**Comment by poeticjustice** [comment]
I can prob set up an additional staging server so you can see it exactly

**Comment by DonnaScriptTechs** [comment]
It's not necessary as I mentioned he's working on an improvement patch. :) He'll put it here for you to test with. It will make it so that one file never gets deleted by the upgrade process as it's a file that shouldn't.

**Comment by poeticjustice** [comment]
3 files you mean  ha :S 

Was he able to replicate this then
@DonnaScriptTechs

**Comment by DonnaScriptTechs** [comment]
I don't know if it's three files but he's working on something for this. As for reproducing, I'm not sure if he can or not but it shouldn't ever delete one of those files.

**Comment by poeticjustice** [comment]
Images incase it helps

![img_2325](https://cloud.githubusercontent.com/assets/19233272/25595307/2deafcd0-2ebc-11e7-96a2-d19e408380f9.PNG)
![img_2326](https://cloud.githubusercontent.com/assets/19233272/25595308/2deefee8-2ebc-11e7-98b4-0b5a1eb4be2f.PNG)
![img_2327](https://cloud.githubusercontent.com/assets/19233272/25595306/2de953d0-2ebc-11e7-894f-93b970e03f54.PNG)
![img_2328](https://cloud.githubusercontent.com/assets/19233272/25595309/2e106ef2-2ebc-11e7-9132-824a6ce89d5b.PNG)

**Comment by RaymondBenc** [comment]
We are testing a fix for future releases, however even with the fix in place for `4.9.*` it won't help anyone upgrading because the flaw seems to be in the package manager. So a patch will need to be applied to the package manager for those that run into this problem.

I've provided a test patch below. This needs to be applied on your server before you attempt to upgrade to 4.9.*.

### Applying the Patch
1. Open the file: `/install/controllers/ManageController.php`
2. Look for...
```
$code = $diffSummary['key'];
```
3. Under that add...
```
if( preg_match('/^(.*?)application\/settings\/([a-z_\.]+)$/', $dest) ) {
   continue;
}
```
4. Save the file and run the upgrade.

### What does the patch do?
The patch simply tells the package manager to skip any sort of interaction with files located in your `application/settings/` folder.

### What is the initial problem?
In my specific case, I had a fresh install of `4.8.13` and in my packages folder I have a file `application/packages/core-base-4.8.11.json` and in that file I had something like...
```
                {
                    "path": "database.php",
                    "dir": false,
                    "file": true,
                    "perms": "0644",
                    "size": 373,
                    "sha1": "eb41a646c9e644f22540b2965f9ef00d4eca44a0"
                },
```
If I remove this before the upgrade, then the upgrade works fine. Not everyone runs into this problem as the files `sha1` or `perms` match and it does nothing to the file.

**Comment by poeticjustice** [comment]
Does it only appear to be 4.8.11 that's affected, because I have ever only installed from 4.8.11

In hindsight should /application/settings/ NOT be set as permission 777 reclusive, all files and folders, as recommended by SE??

**Also another query,** on the page "run preinstall check"

I obviously know some profile fields were fixed, with regards to bugs I reported and which have been fixed which is awesome.

But on the preinstall check.

it doesn't detect them files as being different, as usually, I like to check some files, to compare and see what changes were made.
 
@RaymondBenc

**Comment by DonnaScriptTechs** [comment]
This can be for any previous version. Please let us know how the patch works as it works in our tests. Thank you!

**Comment by gsf00001** [comment]
==>"This can be for any previous version below 4.9.x. "
So, if upgrading from 4.8.13 directly to 4.9.1 (or 4.9.2 directly from 4.8.13), what are the procedures for what to do when?  Thanks.

**Comment by DonnaScriptTechs** [comment]
Do as Ray said and edit the file on your server. You should see that file on your server. After you edit the file, do the upgrade as normal to whichever version. I recommend 4.9.1 as we've not released 4.9.2 yet. You won't need to do this process again for that site.

**Comment by gsf00001** [comment]
Thanks Donna.  I'll pass this info along to FastComet in case they aren't aware of it.  I really don't like muddying up upgrades - there's enough already going on with 4.9.x and 3rd-Party Devs' Plugins.  Oh well.  And I'll be waiting for 4.9.2 (at least) before attempting any of this.

**Comment by DonnaScriptTechs** [comment]
You're welcome @gsf00001 . Please let us know how it works for you.

**Comment by poeticjustice** [comment]
@gsf00001 Im about to test, I have 4 staging servers.

1 for clean install tests
1 for clean install with 3rd party plugins
1 for clean install with 3rd party plugins, with limited interactions and default members
and
1 for current live site clone.

Like to make sure nothing goes wrong lol

**Comment by gsf00001** [comment]
Thanks @poeticjustice .  

Although I won't be doing this (FC will be doing SE 4.9.x, I'll be doing all 3rd-party Plugins or else I may hire each Dev to do their own so there's no fingerpointing and they can resolve any issues at the point they are created), I want to be sure of the exact steps, especially since I'll have Plugins from 6 Devs (only 2 so far are compatible with 4.9.1).

**Comment by poeticjustice** [comment]
As much as Fastcomet are the best, If i was you, Id ask them to clone your site to a staging server in a sub directory, copy the database onto a newly made database by importing, change the mysql config on the staging server to communicate with the clone database............

believe me, if i did all updates on my live site, id be crying right now @gsf00001

**Comment by MMAMELEE** [comment]
@poeticjustice I had this happen to me and it was on my live site! 8 years of info and I didnt have a backup.  I never had a problem before as I've had SE since 3.0 and never had a problem updating even when it was 3.0 to 4.0. I didnt get to read all these comments yet but I hope I can fix it. 
![errorr](https://cloud.githubusercontent.com/assets/12929729/25634905/470f084e-2f4a-11e7-989e-8e705732a214.png)

**Comment by MMAMELEE** [comment]
@poeticjustice did anyone try this patch yet? I really want to get this resolved. I'm scared to even work on the thing now. Thanks for the help guys.

**Comment by MMAMELEE** [comment]
@RaymondBenc what should the permissions be for the install folders? Mine are showing 0644. Keep it at that or change it? Thanks for your time.

**Comment by MMAMELEE** [comment]
@RaymondBenc @poeticjustice do you add the patch code after this red or blue line? Is this the patch for the same issue I posted before? Thanks
![code](https://cloud.githubusercontent.com/assets/12929729/25635565/96ee4ada-2f4c-11e7-9897-e0b84fadaabd.png)

**Comment by gsf00001** [comment]
@poeticjustice Good advice.

I'm actually a backup/clone fanatic (been that way for over 30 years).  This is one of the reasons I don't like auto-backups replacing original file - I'll chose what to backup when.  There are typically multiple folders for each clone (prod, stg, test, live - even though I'm not live yet, I do it for the sake of procedure).  Plus redundant backups at both primary host site and secondary .ftp site (not affiliate with primary host site).  And don't get me started on the physical backup routines for certain things :)

  It's like a seatbelt - I may not have needed one in 25 years but I still don't move my car without it on.

**Comment by MMAMELEE** [comment]
@RaymondBenc @poeticjustice @DonnaScriptTechs my initial error for my update said this 
![errorupdating](https://cloud.githubusercontent.com/assets/12929729/25635629/cb7fe61e-2f4c-11e7-9395-e050560154c2.png)  I changed those now I have this in the manage update
![errorr](https://cloud.githubusercontent.com/assets/12929729/25635651/dcb4d444-2f4c-11e7-9613-03ddf5c874be.png)  Will doing your patch work to fix mine? I also am not sure where to add the code 
![code](https://cloud.githubusercontent.com/assets/12929729/25635667/ec59e588-2f4c-11e7-99ac-d67743691b31.png)

**Comment by MMAMELEE** [comment]
Sorry I just noticed it was hard to read I will do it separate  @RaymondBenc @DonnaScriptTechs @Elshara @poeticjustice . My initial error for my update said this
![errorupdating](https://cloud.githubusercontent.com/assets/12929729/25635728/2bd1ddce-2f4d-11e7-8c04-ca074c02c9a8.png)

**Comment by MMAMELEE** [comment]
Then I changed the permissions the site came up but not correct and it shows this 
![errorr](https://cloud.githubusercontent.com/assets/12929729/25635752/3d04a32e-2f4d-11e7-93b2-433e6050e12d.png)

**Comment by MMAMELEE** [comment]
@RaymondBenc @DonnaScriptTechs sorry for all the tags I just want to be able to get this correct. Will this patch work for me? Also my file permissions in the manageinstall.php are 0644. Should I change them? 
![code](https://cloud.githubusercontent.com/assets/12929729/25635798/66dc6682-2f4d-11e7-97de-3bf6f6493a0a.png)

**Comment by DonnaScriptTechs** [comment]
@MMAMELEE please send a ticket to our support for assistance if you have support. If you don't, you'll need to purchase it. You might need your host to restore the site if you don't have a backup and if the database was deleted.

**Comment by DonnaScriptTechs** [comment]
Oh I see you say above that it wasn't deleted. That's good. Please do contact our support for assistance with the patch if you need it. You will need active support.

**Comment by DonnaScriptTechs** [comment]
Just to clarify the issue a little, the database isn't actually deleted. The setting file does and makes it seem like the database is deleted.

**Comment by poeticjustice** [comment]
@DonnaScriptTechs correct it doesn't delete the database
However if you see the install page because it deletes the files.
Then if you go thru the install process with SE product key, community name and admin email etc

It will wipe all user data, that's what happened to me, it wiped all MySQL database entries for members, over a thousand people.

Only because after update it didn't have them files as deleted, so once put data on for the database, instead of instal going "thanks it seems database is already established"
It continues to set up the Social Network from scratch. And it wipes and overwrites the MySQL enteries...... so technically it does wipe the most important parts of the DB.


Yet to try new patch as internet is doen

**Comment by DonnaScriptTechs** [comment]
Please let us know if the patch works once you test it. Thank you!

**Comment by MMAMELEE** [comment]
@DonnaScriptTechs @RaymondBenc will the patch work with my issue?

**Comment by poeticjustice** [comment]
@MMAMELEE @DonnaScriptTechs im testing now

**Comment by DonnaScriptTechs** [comment]
Thank you @poeticjustice . 

@MMAMELEE it should work but you can contact our support for more assistance if you need it. You'll need an active support plan for that. Your file permissions should be as the article, http://support.socialengine.com/php/customer/en/portal/articles/2777304-optimized-file-and-folder-permissions?b_id=14386

**Comment by MMAMELEE** [comment]
@poeticjustice yes please let me know when you know...thank you. Also can you tell me what will delete database? Also @DonnaScriptTechs and @RaymondBenc you never answered about the memberinstall.php permissions...mine are at 0644 do they have to be at something different? Thank you

**Comment by DonnaScriptTechs** [comment]
The link I provided above has all of the permissions listed. If you follow that, it will have the correct permissions. Some of the instructions mention to do the permissions recursively. Please contact our support for further assistance as we cannot provide support assistance through our Github channel. Thank you!

**Comment by MMAMELEE** [comment]
Sorry @DonnaScriptTechs I didn't see your link before. Thank you

**Comment by DonnaScriptTechs** [comment]
@MMAMELEE no problem at all. :)

**Comment by DonnaScriptTechs** [comment]
@poeticjustice how did your testing go?

**Comment by poeticjustice** [comment]
Sorry ive been really Ill last few days, Im litrally trying this all now, update you soon

**Comment by poeticjustice** [comment]
Hi, The patch seems to have worked,

It still said 
/application/settings/database.php (different_removed)
/application/settings/enabled_module_directories.php (different_removed)
/application/settings/general.php (different_removed)

and /install/controllers/ManageController.php (different_different)

yet it seems it didnt actually delete the first 3 even tho it said different_removed)

It did however overwrite /install/controllers/ManageController.php 


It says its updated to 4.9.1

seems to be something weird in the notifications page 

tho, like an image that is small repeated, over and over again, i cleared my browser cache and its still there


![image1](https://cloud.githubusercontent.com/assets/19233272/25831754/1ec257b0-345e-11e7-93dd-2a455ce54eb8.JPG)

**Comment by DonnaScriptTechs** [comment]
That notification issue is not related to this. It is in another bug report that was deemed a third party issue. You would want to check third party products. Make sure you have them all updated. https://github.com/SocialEngine/phpv4-issues/issues/660

Thanks for confirming the fix works.

**Comment by poeticjustice** [comment]
ok thanks @DonnaScriptTechs if you get time can you check my post on diff checks on onsocial community

**Comment by MMAMELEE** [comment]
@DonnaScriptTechs & @poeticjustice I have that problem too on a bunch of spots I saw and I dont have 3rd party plugins
![nopf1](https://cloud.githubusercontent.com/assets/12929729/25973086/aba0773e-3670-11e7-9d4b-1335f38406c3.png)

**Comment by MMAMELEE** [comment]
Also @DonnaScriptTechs @RaymondBenc you never responded on where to enter the code on the manageinstal.php - All the profiles on my site seem to work but the one I'm signed into it always gives  me this error when I try to view my profile 
![nopf](https://cloud.githubusercontent.com/assets/12929729/25973141/db9ee736-3670-11e7-9cdf-0aca9cacc94e.png)

**Comment by MMAMELEE** [comment]
@DonnaScriptTechs @poeticjustice more examples of what you showed
![nopf1](https://cloud.githubusercontent.com/assets/12929729/25973198/06259932-3671-11e7-8d71-e9ce67c5a446.png)

**Comment by DonnaScriptTechs** [comment]
@MMAMELEE we mentioned to please contact our support as we can't use github as a support portal for your issues. The issues you are showing for the icons are not related to this issue report. Please use a support ticket for assistance. 

As this works and will be implemented in our 4.9.2 release, I'll close it now.
